### PR TITLE
Fix discovery packet size

### PR DESCRIPTION
**Pull request by atoulme**
Signed-off-by: Antoine Toulme <antoine@lunar-ocean.com>

## PR description
Make sure Besu sends NEIGHBORS packets under MTU size.

## Fixed Issue(s)
#2250 

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java [hunk 1]
```diff
@@ -224,19 +224,19 @@ public void neighborsPacketLimited() {
...
     assertThat(neighborsPacket.fromAgent).isEqualTo(agent);
 
-    // Assert that we only received 16 items.
+    // Assert that we only received 13 items.
     assertThat(neighborsPacket.packet.getPacketData(NeighborsPacketData.class).isPresent())
         .isTrue();
     final NeighborsPacketData neighbors =
         neighborsPacket.packet.getPacketData(NeighborsPacketData.class).get();
     assertThat(neighbors).isNotNull();
-    assertThat(neighbors.getNodes()).hasSize(16);
```
Comment (shemnon) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java:233]: If the purpose of the change is to ensure packet size on the wire is < 1144 bytes then the unit test should test the encoded packet size.

Comment (atoulme) [reply ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java:233]: Added something, though this test doesn't encode packets, bypassing the whole UDP transport.

**Review by shemnon — APPROVED** [review]

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java [hunk 2]
```diff
@@ -224,19 +224,20 @@ public void neighborsPacketLimited() {
...
+    assertThat(neighbors.getNodes()).hasSize(13);
+    assertThat(neighborsPacket.packet.encode().length()).isLessThanOrEqualTo(1280); // under max MTU
 
-    // Assert that after removing those 16 items we're left with either 4 or 5.
+    // Assert that after removing those 13 items we're left with either 4 or 5.
```
Comment (shemnon) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java:236]: nit.  7 or 8
```suggestion
    // Assert that after removing those 13 items we're left with either 7 or 8.
```

File: ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java [hunk 3]
```diff
@@ -224,19 +224,20 @@ public void neighborsPacketLimited() {
...
     // If we are left with 5, the test peer was returned as an item, assert that this is the case.
```
Comment (shemnon) [inline ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgentTest.java:237]: nit, 8
```suggestion
    // If we are left with 8, the test peer was returned as an item, assert that this is the case.
```

**Pull request merged**